### PR TITLE
Fix loop over diff content

### LINKED ISSUE #13711: [production][request#show] undefined method `each_line' for nil:NilClass

    @content.each_line.with_index do |line, line_i...
[See this exception on Errbit](http://errbit.opensuse.org/apps/5620ca2bdc71fa00b1000000/problems/63c6ae2a84bf466764ed6252)
# undefined method `each_line&#39; for nil:NilClass

    @content.each_line.with_index do |line, line_index|
            ^^^^^^^^^^ #
### URL ###
[https://build.opensuse.org/request/show/1058888](https://build.opensuse.org/request/show/1058888)"
### Where ###
request#show

### Occured ###
Jan 17  2:18pm

### Similar ###
0

## Params ##
~~~
{
  &quot;action&quot; =&gt; &quot;show&quot;,
  &quot;controller&quot; =&gt; &quot;webui/request&quot;,
  &quot;number&quot; =&gt; &quot;1058888&quot;
}
~~~

## Session ##
~~~
{
  &quot;_csrf_token&quot; =&gt; &quot;hcGhR7ThTwFzfqyFCL8pmN-hAmmTRAKRfgcR-820PTs&quot;,
  &quot;session_id&quot; =&gt; &quot;fbda46bd4418729eb0b7c8478d2484d4&quot;
}
~~~

## Backtrace ##
~~~
12:  app/services/diff_parser.rb -> **call**
11:  app/services/action_sourcediff_parser.rb -> **block (2 levels) in call**
8:  app/services/action_sourcediff_parser.rb -> **each**
8:  app/services/action_sourcediff_parser.rb -> **block in call**
7:  app/services/action_sourcediff_parser.rb -> **each**
7:  app/services/action_sourcediff_parser.rb -> **call**
15:  app/components/sourcediff_component.rb -> **initialize**
2:  app/views/webui/request/beta_show_tabs/_changes.html.haml -> **new**
2:  app/views/webui/request/beta_show_tabs/_changes.html.haml -> **_app_views_webui_request_beta_show_tabs__changes_html_haml___1023393746690362048_101780**
244:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/base.rb -> **public_send**
244:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/base.rb -> **_run**
157:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/template.rb -> **block in render**
208:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **instrument**
361:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/template.rb -> **instrument_render_template**
155:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/template.rb -> **render**
251:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/partial_renderer.rb -> **block in render_partial_template**
206:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **block in instrument**
24:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications/instrumenter.rb -> **instrument**
206:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **instrument**
246:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/partial_renderer.rb -> **render_partial_template**
237:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/partial_renderer.rb -> **render**
81:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/renderer.rb -> **render_partial_to_object**
27:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/renderer.rb -> **render_to_object**
22:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/renderer.rb -> **render**
37:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/helpers/rendering_helper.rb -> **block in render**
270:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/base.rb -> **in_rendering_context**
33:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/helpers/rendering_helper.rb -> **render**
103:  app/views/webui/request/beta_show.html.haml -> **_app_views_webui_request_beta_show_html_haml__584487688546552533_101220**
244:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/base.rb -> **public_send**
244:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/base.rb -> **_run**
157:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/template.rb -> **block in render**
208:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **instrument**
361:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/template.rb -> **instrument_render_template**
155:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/template.rb -> **render**
65:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/template_renderer.rb -> **block (2 levels) in render_template**
206:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **block in instrument**
24:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications/instrumenter.rb -> **instrument**
206:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **instrument**
60:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/template_renderer.rb -> **block in render_template**
75:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/template_renderer.rb -> **block in render_with_layout**
208:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **instrument**
74:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/template_renderer.rb -> **render_with_layout**
59:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/template_renderer.rb -> **render_template**
11:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/template_renderer.rb -> **render**
61:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/renderer.rb -> **render_template_to_object**
29:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/renderer/renderer.rb -> **render_to_object**
117:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/rendering.rb -> **block in _render_template**
270:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/base.rb -> **in_rendering_context**
116:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/rendering.rb -> **_render_template**
216:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/streaming.rb -> **_render_template**
103:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/rendering.rb -> **render_to_body**
46:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/rendering.rb -> **render_to_body**
141:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/renderers.rb -> **render_to_body**
25:  /GEM_ROOT/gems/actionpack-7.0.4/lib/abstract_controller/rendering.rb -> **render**
30:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/rendering.rb -> **render**
22:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/instrumentation.rb -> **block (2 levels) in render**
311:  /usr/lib64/ruby/3.1.0/benchmark.rb -> **realtime**
14:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/core_ext/benchmark.rb -> **ms**
22:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/instrumentation.rb -> **block in render**
91:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/instrumentation.rb -> **cleanup_view_runtime**
34:  /GEM_ROOT/gems/activerecord-7.0.4/lib/active_record/railties/controller_runtime.rb -> **cleanup_view_runtime**
21:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/instrumentation.rb -> **render**
67:  app/controllers/webui/request_controller.rb -> **show**
6:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/basic_implicit_render.rb -> **send_action**
215:  /GEM_ROOT/gems/actionpack-7.0.4/lib/abstract_controller/base.rb -> **process_action**
53:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/rendering.rb -> **process_action**
234:  /GEM_ROOT/gems/actionpack-7.0.4/lib/abstract_controller/callbacks.rb -> **block in process_action**
107:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/callbacks.rb -> **run_callbacks**
233:  /GEM_ROOT/gems/actionpack-7.0.4/lib/abstract_controller/callbacks.rb -> **process_action**
22:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/rescue.rb -> **process_action**
67:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/instrumentation.rb -> **block in process_action**
206:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **block in instrument**
24:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications/instrumenter.rb -> **instrument**
206:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/notifications.rb -> **instrument**
66:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/instrumentation.rb -> **process_action**
259:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal/params_wrapper.rb -> **process_action**
27:  /GEM_ROOT/gems/activerecord-7.0.4/lib/active_record/railties/controller_runtime.rb -> **process_action**
151:  /GEM_ROOT/gems/actionpack-7.0.4/lib/abstract_controller/base.rb -> **process**
39:  /GEM_ROOT/gems/actionview-7.0.4/lib/action_view/rendering.rb -> **process**
188:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal.rb -> **dispatch**
251:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_controller/metal.rb -> **dispatch**
49:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/routing/route_set.rb -> **dispatch**
32:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/routing/route_set.rb -> **serve**
18:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/routing/mapper.rb -> **block in &lt;class:Constraints&gt;**
48:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/routing/mapper.rb -> **serve**
50:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/journey/router.rb -> **block in serve**
32:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/journey/router.rb -> **each**
32:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/journey/router.rb -> **serve**
852:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/routing/route_set.rb -> **call**
72:  /GEM_ROOT/gems/flipper-0.26.0/lib/flipper/middleware/memoizer.rb -> **memoized_call**
37:  /GEM_ROOT/gems/flipper-0.26.0/lib/flipper/middleware/memoizer.rb -> **call**
15:  config/initializers/prefer_xml_over_html.rb -> **call**
44:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/deflater.rb -> **call**
15:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/tempfile_reaper.rb -> **call**
27:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/etag.rb -> **call**
27:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/conditional_get.rb -> **call**
12:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/head.rb -> **call**
38:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/http/permissions_policy.rb -> **call**
36:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/http/content_security_policy.rb -> **call**
266:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/session/abstract/id.rb -> **context**
260:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/session/abstract/id.rb -> **call**
696:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/cookies.rb -> **call**
27:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/callbacks.rb -> **block in call**
99:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/callbacks.rb -> **run_callbacks**
26:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/callbacks.rb -> **call**
32:  /GEM_ROOT/gems/airbrake-9.4.3/lib/airbrake/rack/middleware.rb -> **call!**
21:  /GEM_ROOT/gems/airbrake-9.4.3/lib/airbrake/rack/middleware.rb -> **call**
28:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/debug_exceptions.rb -> **call**
26:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/show_exceptions.rb -> **call**
18:  /GEM_ROOT/gems/lograge-0.12.0/lib/lograge/rails_ext/rack/logger.rb -> **call_app**
25:  /GEM_ROOT/gems/railties-7.0.4/lib/rails/rack/logger.rb -> **block in call**
99:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/tagged_logging.rb -> **block in tagged**
37:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/tagged_logging.rb -> **tagged**
99:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/tagged_logging.rb -> **tagged**
25:  /GEM_ROOT/gems/railties-7.0.4/lib/rails/rack/logger.rb -> **call**
93:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/remote_ip.rb -> **call**
93:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/remote_ip.rb -> **call**
19:  /GEM_ROOT/gems/request_store-1.5.1/lib/request_store/middleware.rb -> **call**
26:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/request_id.rb -> **call**
24:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/method_override.rb -> **call**
22:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/runtime.rb -> **call**
29:  /GEM_ROOT/gems/activesupport-7.0.4/lib/active_support/cache/strategy/local_cache_middleware.rb -> **call**
14:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/executor.rb -> **call**
18:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/lock.rb -> **call**
110:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/sendfile.rb -> **call**
131:  /GEM_ROOT/gems/actionpack-7.0.4/lib/action_dispatch/middleware/host_authorization.rb -> **call**
530:  /GEM_ROOT/gems/railties-7.0.4/lib/rails/engine.rb -> **call**
226:  /GEM_ROOT/gems/railties-7.0.4/lib/rails/railtie.rb -> **public_send**
226:  /GEM_ROOT/gems/railties-7.0.4/lib/rails/railtie.rb -> **method_missing**
74:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/urlmap.rb -> **block in call**
58:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/urlmap.rb -> **each**
58:  /GEM_ROOT/gems/rack-2.2.4/lib/rack/urlmap.rb -> **call**
107:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb -> **process_request**
149:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb -> **accept_and_process_next_request**
110:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb -> **main_loop**
419:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/request_handler.rb -> **block (3 levels) in start_threads**
113:  /usr/lib64/ruby/gems/3.1.0/gems/passenger-6.0.14/src/ruby_supportlib/phusion_passenger/utils.rb -> **block in create_thread_and_abort_on_exception**
~~~

## Environment ##
| Key        | Value      |
|------------|------------|
| program_name | Passenger RubyApp: /srv/www/obs/api (production) |
| HTTP_USER_AGENT | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36 Edg/109.0.1518.55 |

### PR DESCRIPTION
**Pull request by eduardoj**
Fixes #13711.

### CODE REVIEW

File: src/api/app/services/diff_parser.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def initialize(content: [])
   end
 
   def call
-    @content.each_line.with_index do |line, line_index|
+    @content&.each_line&.with_index do |line, line_index|
```
Comment (hennevogel) [inline src/api/app/services/diff_parser.rb:12]: Hm, one should be enough no? How can this be nil if it's required during `initialize`?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/app/services/diff_parser.rb (hunk 1 above)
Comment (eduardoj) [reply src/api/app/services/diff_parser.rb:12]: I received a message from RuboCop: `Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.`. This was the automatic fix.

Maybe it is not only `nil`...

Comment (hennevogel) [reply src/api/app/services/diff_parser.rb:12]: Maybe? We should know no?

**Review by danidoni — CHANGES_REQUESTED** [review]

File: src/api/app/services/diff_parser.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class DiffParser
-  def initialize(content: [])
+  def initialize(content:)
     @content = content
```
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/services/diff_parser.rb:3]: Then you can remove the `&.` operators

[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/services/diff_parser.rb]: Yes, let's try this. Thanks!

File: src/api/app/services/diff_parser.rb (hunk 1 above)
Comment (danidoni) [reply src/api/app/services/diff_parser.rb]: Providing the default value in the constructor we do not need to worry about this anymore

**Review by hennevogel — APPROVED** [review]

File: src/api/app/services/diff_parser.rb (hunk 1 above)
Comment (eduardoj) [reply src/api/app/services/diff_parser.rb]: We didn't take in account `content` being `nil`. This is now fixed, and a test case added. Please have a look again.

**Review by danidoni — APPROVED** [review]

File: src/api/app/services/diff_parser.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (danidoni) [reply src/api/app/services/diff_parser.rb]: I realize content is a string, not a collection!

**Pull request merged**